nsdb-core/src/main/scala/io/radicalbit/nsdb/statement/ExpressionParser.scala

```diff
@@ -40,14 +40,32 @@ object ExpressionParser {
     val q = exp match {
       case Some(NullableExpression(dimension)) => nullableExpression(schema, dimension)
 
-      case Some(EqualityExpression(dimension, value)) => equalityExpression(schema, dimension, value)
+      case Some(EqualityExpression(dimension, value)) =>
+        value match {
+          case AbsoluteComparisonValue(value)          => equalityExpression(schema, dimension, value)
```

Comment:
suspicious shadowing here?

Reply:
@saveveltri what do you think about it?

Reply:
maybe change the `value ` inside `AbsoluteComparisonValue(value)` (and the other) with another name like `v` or something else?

Reply:
yes. good catch @riccardo14.
In this case, since we return the same result, no matter the actual implementation, we can use the unapply of the object ComparisonValue in the case at line 43, in order to deconstruct the value and use it once